akka-docs/src/main/paradox/typed/style-guide.md

```diff
@@ -295,3 +300,151 @@ Scala
+
+Java
+:  @@snip [StyleGuideDocExamples.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/StyleGuideDocExamples.java) { #on-message-lambda-anti }
+
+In a real application it would often be more than 3 lines for each message.
+It's not only making it more difficult to get an overview of the message matching, but compiler errors related
+to lambdas can sometimes be difficult to understand.
+
+Ideally, lambdas should be written in one line of code. Two lines can be ok, but three is probably too much.
+Also, don't use braces and return statements in one-line lambda bodies.
+
+@@@
+
+@@@ div {.group-scala}
+
+## Partial vs. total Function
+
+It's recommended to use a `sealed` trait as the super type of the commands (incoming messages) of a an actor
+because then the Scala compiler will emit a warning if a message type is forgotten in the pattern match.
+
+Scala
+:  @@snip [StyleGuideDocExamples.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala) { #messages-sealed }
+
+That is the main reason for why `Behaviors.receive`, `Behaviors.receiveMessage` takes a total `Function` and
+not a `PartialFunction`.
+
+The compiler warning if `GetValue` is not handled:
+
+```
+[warn] ... Counter.scala:45:34: match may not be exhaustive.
+[warn] It would fail on the following input: GetValue(_)
+[warn]         Behaviors.receiveMessage {
+[warn]                                  ^
+```
+
+Note that a `MatchError` will be thrown in runtime if a message is not handled, so it's important to pay
+attention to those. If a `Behavior` should not handle certain messages you can still include them
+in the pattern match and return `Behaviors.unhandled`.
+
+Scala
+:  @@snip [StyleGuideDocExamples.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala) { #pattern-match-unhandled }
+
+One thing to be aware of is that the exhaustiveness check is not enabled when there is a guard condition in the
+pattern match cases.
+
+Scala
+:  @@snip [StyleGuideDocExamples.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala) { #pattern-match-guard }
+
+Therefore it can be better to not use the guard and instead move the `if` after the `=>`.
+
+Scala
+:  @@snip [StyleGuideDocExamples.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala) { #pattern-match-without-guard }
+
+It's recommended to use the `sealed` trait and total functions with exhaustiveness check to detect mistakes
+of forgetting to handle some messages. Sometimes that can be inconvenient and then you can use a `PartialFunction`
+with `Behaviors.receivePartial` or `Behaviors.receiveMessagePartial`
+
+Scala
+:  @@snip [StyleGuideDocExamples.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala) { #pattern-match-partial }
+
+@@@
+
+@@@ div {.group-scala}
+
+## ask vs ?
```

Comment:
```suggestion
## `ask` versus `?`
```

Reply:
yeah, I don't think we should have formatting in headings